internal/services/apicenter/api_center_environment_resource_test.go

```diff
@@ -0,0 +1,183 @@
+
+package apicenter_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/apicenter/2024-03-01/environments"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+type ApiCenterEnvironmentResource struct{}
+
+func TestAccApiCenterEnvironment_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_api_center_environment", "test")
+	r := ApiCenterEnvironmentResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccApiCenterEnvironment_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_api_center_environment", "test")
+	r := ApiCenterEnvironmentResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.update(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccApiCenterEnvironment_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_api_center_environment", "test")
+	r := ApiCenterEnvironmentResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func (ApiCenterEnvironmentResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := environments.ParseEnvironmentID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clients.ApiCenter.EnvironmentsClient.Get(ctx, *id)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+
+	return pointer.To(resp.Model != nil), nil
+}
+
+func (ApiCenterEnvironmentResource) template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-apicenter-%d"
+  location = "%s"
+}
+
+resource "azurerm_api_center_service" "test" {
+  name                = "acctestApiCSvc%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+
+  identity {
+    type = "SystemAssigned"
+  }
+}
+
+`, data.RandomInteger, "eastus", data.RandomInteger) // Only available in a few select regions for now
```

Comment:
The location override should be done in the [TC config](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/.teamcity/components/settings.kt)